build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -53,301 +53,304 @@ let wrapCounter = 0;
  */
 let transform;
 
-/**
- * Updates the set of preprocessors to run on HTML and JS files before testing.
- * Notes:
- * - The HTML transform is lazy-required because the server is built at startup.
- * - We must use babel on windows until esbuild can natively downconvert to ES5.
- * @param {!RuntimeTestConfig} config
- */
-function updatePreprocessors(config) {
-  const createHtmlTransformer = function () {
-    return function (content, _file, done) {
-      if (!transform) {
-        const outputDir = `../../../${SERVER_TRANSFORM_PATH}/dist/transform`;
-        transform = require(outputDir).transformSync;
-      }
-      done(transform(content));
-    };
-  };
-  createHtmlTransformer.$inject = [];
-  config.plugins.push({
-    'preprocessor:htmlTransformer': ['factory', createHtmlTransformer],
-  });
-  config.preprocessors[karmaHtmlFixturesPath] = ['htmlTransformer', 'html2js'];
-  for (const karmaJsPath of karmaJsPaths) {
-    config.preprocessors[karmaJsPath] = ['esbuild'];
+class RuntimeTestConfig {
+  /** @type {Array<string|[]>} */
+  plugins = [];
+
+  /**@type {Record<string, string|string[]>} */
+  preprocessors = {};
+
+  /** @type {string[]} */
+  reporters = [];
+
+  client = {};
```

Comment:
While this works as a way to silence existing type errors, it feels like we're having to declare an arbitrary subset of fields from `karmaConfig`. Curious about why the others aren't needed.

(Okay to do this for now and refine later if an alternative can be identified.)

Reply:
I left a comment explaining this, but its definitely going to stay on my mind as something to potentially be cleaned up.